spi/src/main/java/org/opensearch/jobscheduler/spi/utils/JobHistoryService.java

```diff
@@ -0,0 +1,261 @@
+import org.opensearch.common.xcontent.LoggingDeprecationHandler;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.index.engine.VersionConflictEngineException;
+import org.opensearch.index.seqno.SequenceNumbers;
+import org.opensearch.action.admin.indices.create.CreateIndexRequest;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.xcontent.MediaType;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.jobscheduler.spi.StatusHistoryModel;
+import org.opensearch.transport.client.Client;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.Map;
+
+public class JobHistoryService {
+    private static final Logger logger = LogManager.getLogger(JobHistoryService.class);
+    public static final String JOB_HISTORY_INDEX_NAME = ".opendistro-job-scheduler-history";
+
+    private final Client client;
+    private final ClusterService clusterService;
+    final static Map<String, Object> INDEX_SETTINGS = Map.of("index.number_of_shards", 1, "index.auto_expand_replicas", "0-all");
+
+    public JobHistoryService(final Client client, final ClusterService clusterService) {
+        this.client = client;
+        this.clusterService = clusterService;
+    }
+
+    private String historyMapping() {
+        try {
+            InputStream in = JobHistoryService.class.getResourceAsStream("opensearch_job_scheduler_history.json");
+            StringBuilder stringBuilder = new StringBuilder();
+            BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8));
+            for (String line; (line = bufferedReader.readLine()) != null;) {
+                stringBuilder.append(line);
+            }
+            return stringBuilder.toString();
+        } catch (IOException e) {
+            throw new IllegalArgumentException("History Mapping cannot be read correctly.");
+        }
+    }
+
+    public boolean historyIndexExist() {
+        return clusterService.state().routingTable().hasIndex(JOB_HISTORY_INDEX_NAME);
+    }
+
+    void createHistoryIndex(ActionListener<Boolean> listener) {
+        try (ThreadContext.StoredContext ignore = client.threadPool().getThreadContext().stashContext()) {
+            if (historyIndexExist()) {
+                listener.onResponse(true);
+            } else {
+                final CreateIndexRequest request = new CreateIndexRequest(JOB_HISTORY_INDEX_NAME).mapping(
+                    historyMapping(),
```

Comment:
I think its fine to declare the mappings, but I'm dealing with a problem in ISM that also creates mappings on index creation and then does not allow new mappings to be added in subsequent minor versions. 

Let's keep this here, but if we were to add a new field (say `exception_message`) then how would that work on upgrade?